jersey-servers/src/main/java/com/palantir/remoting2/servers/jersey/TraceEnrichingFilter.java

```diff
@@ -91,4 +98,24 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
             return Optional.of(header.equals("1"));
         }
     }
+
+    private static void addUserIdMetadataToSpan(ContainerRequestContext requestContext) {
+        Optional<String> userId = getUserIdFromAuthHeader(requestContext);
+        if (userId.isPresent()) {
+            Tracer.addSpanMetadata(ImmutableMap.of(USER_ID_METADATA_KEY, userId.get()));
+        }
+    }
+
+    private static Optional<String> getUserIdFromAuthHeader(ContainerRequestContext requestContext) {
```

Comment:
we're already doing this processing to populate the MDC and the request context, probably ought to revisit how the filters end up populating these various threadlocals. The processing isn't free and these are per-request overheads.